### ISSUE TITLE
Malloced regions in x86 application are not mapped in memory during debugging (elaboration of #3641)

### ISSUE THREAD

**Issue by Smileynator** [NationalSecurityAgency/ghidra#3676]
**Describe the bug**
Officially marking this as a bug after brief discussion on the topic. I am pretty sure it's not on my end.
I can't seem to get any malloced memory addresses (between 0x0200000 and 0x0400000 ish, exact values change per session ofcourse) to be accessible at any point. Which makes actually inspecting contents a lot harder to do.
I tried my best on including all the relevant information i could find, from 1 session. I included only 1 screen of actual memory not being accessible, but every instance of this, and many other malloced structs of many different types show the same behavior.

**To Reproduce**
1. debug X86 application
2. breakpoint at malloc
3. try to navigate towards the returned pointer and fail

**Expected behavior**
See memory on the heap properly.

**Screenshots**
![image](https://user-images.githubusercontent.com/3433068/143500028-47a32b0f-1740-4a42-87e8-213186d9b36a.png)
![image](https://user-images.githubusercontent.com/3433068/143500051-f2e7038b-7cde-4840-af5e-09ed41d711fa.png)
![image](https://user-images.githubusercontent.com/3433068/143500169-d787d411-cd39-43ba-bfcb-752fa4c43e8b.png)
![image](https://user-images.githubusercontent.com/3433068/143500187-7bfda613-824b-4b71-8dec-15953add695c.png)
![image](https://user-images.githubusercontent.com/3433068/143500255-98e15d65-865f-48ea-97a3-df5ba54920d8.png)
![image](https://user-images.githubusercontent.com/3433068/143500284-d5db7a52-927f-48e6-b8d2-de47f0cafeb2.png)

**Environment (please complete the following information):**
 - OS: Win 10 latest patches
 - Java Version: 11.0.8
 - Ghidra Version: 10.0.4 & 10.1_BETA
 - Ghidra Origin: Github release page zip file

**Additional context**
Related to my original questions posted here:
https://github.com/NationalSecurityAgency/ghidra/issues/3667

**Comment by d-millar** [comment]
Still looking at this, but I think the issue is that the Memory doesn't auto-refresh for Windows.  Try selecting Memory in the Objects window and hitting refresh (maybe more than once :( ) after the new/malloc has executed.  !address in the Interpreter, the regions listed under Memory in Objects, and the Regions window should agreed (minus FREE regions which are not recorded the Objects/Regions windows).  We used to auto-refresh in single-step but the performance of the steppong was going to seed pretty quickly.

**Comment by d-millar** [comment]
P.S. I realize this is frustrating, but, if we could keep this discussion to one issue rather than opening new ones, it would help us (and probably anyone else having these issues) immensely.

**Comment by Smileynator** [comment]
> P.S. I realize this is frustrating, but, if we could keep this discussion to one issue rather than opening new ones, it would help us (and probably anyone else having these issues) immensely.

Sorry about that, but my question threads turned into more topics than one. And some turned out to be actual issues. I figured i would separate the information per topic where i was certain on it, for documentation sake.

As for the refreshing, i think i tried yesterday with no results. But if refreshing is the solution (i will try again a bit more aggressively this evening) that is at least a workable solution for the time being if i can get that to work.

Auto refreshing per step seems like a terrible plan. But at least one on breakpoint hit might be helpful, and if it really is terrible performance, make it an optional thing, or a dialog when you try to go to memory that isn't recognized "want to rescan memory in order to resolve?" could be the quick fix that is clear to end users?

**Comment by d-millar** [comment]
Agreed - I think we need some sort of better solution.  @nsadeveloper789 has added a feature to force all memory in, which should come out with the next release, but your issues make me think we should also update memory on "Go" and ask the user if he/she'd like memory forced in if it still fails.

**Comment by Smileynator** [comment]
@d-millar To my surprise, you were right with "maybe a few times". I clicked the button to refresh memory, nothing happened. Opened the memory window to see if it even refreshed, clicked again. Did visually refresh, nothing changed.
Aggressively spamclick the button 4 or 5 times, and suddenly a ton more addresses get mapped and i have my missing heap allocations up until this point into the program.

While less than ideal for now, this is workable, thanks for the tip!

**Comment by d-millar** [comment]
@Smileynator Definitely NOT IDEAL, but glad it worked - definitely in my queue!
